tests/ssh_test_utils.py

```diff
@@ -1,55 +1,206 @@
+        central_path=f"/home/sshuser/datashuttle/{project.project_name}",
+        central_host_id="localhost",
+        central_host_username="sshuser",
     )
-    project.update_config_file(central_host_id=central_host_id)
-    project.update_config_file(central_host_username=central_host_username)
-    project.update_config_file(connection_method="ssh")
-
     rclone.setup_rclone_config_for_ssh(
         project.cfg,
         project.cfg.get_rclone_config_name("ssh"),
         project.cfg.ssh_key_path,
     )
 
 
-def setup_mock_input(input_):
+def setup_ssh_connection(project, setup_ssh_key_pair=True):
     """
-    This is very similar to pytest monkeypatch but
-    using that was giving me very strange output,
-    monkeypatch.setattr('builtins.input', lambda _: "n")
-    i.e. pdb went deep into some unrelated code stack
+    Convenience function to verify the server hostkey and ssh
+    key pairs to the Dockerfile image for ssh tests.
+
+    This requires monkeypatching a number of functions involved
+    in the SSH setup process. `input()` is patched to always
+    return the required hostkey confirmation "y". `getpass()` is
+    patched to always return the password for the container in which
+    SSH tests are run. `isatty()` is patched because when running this
+    for some reason it appears to be in a TTY - this might be a
+    container thing.
     """
+    # Monkeypatch
     orig_builtin = copy.deepcopy(builtins.input)
-    builtins.input = lambda _: input_  # type: ignore
-    return orig_builtin
+    builtins.input = lambda _: "y"  # type: ignore
+
+    orig_getpass = copy.deepcopy(ssh.getpass.getpass)
+    ssh.getpass.getpass = lambda _: "password"  # type: ignore
+
+    orig_isatty = copy.deepcopy(sys.stdin.isatty)
+    sys.stdin.isatty = lambda: True
+
+    # Run setup
+    verified = ssh.verify_ssh_central_host(
+        project.cfg["central_host_id"], project.cfg.hostkeys_path, log=True
+    )
+
+    if setup_ssh_key_pair:
+        ssh.setup_ssh_key(project.cfg, log=False)
+
+    # Restore functions
+    builtins.input = orig_builtin
+    ssh.getpass.getpass = orig_getpass
+    sys.stdin.isatty = orig_isatty
+
+    return verified
 
 
-def restore_mock_input(orig_builtin):
+def setup_ssh_container(container_name):
     """
-    orig_builtin: the copied, original builtins.input
+    Build and run the docker container used for
+    ssh tests.
     """
-    builtins.input = orig_builtin
+    assert docker_is_running(), (
+        "docker is not running, "
+        "this should be checked at the top of test script"
+    )
+
+    image_path = Path(__file__).parent / "ssh_test_images"
+    os.chdir(image_path)
+
+    if platform.system() != "Windows":
+        build_command = "sudo docker build -t ssh_server ."
+        run_command = (
+            f"sudo docker run -d -p {PORT}:22 "
+            f"--name {container_name} ssh_server"
+        )
+    else:
+        build_command = "docker build ."
```

Comment:
```suggestion
    else:
        build_command = "docker build -t ssh_server ."
```

We should tag the docker image so that `docker run` can work.